### PR TITLE
[SPARK-28051][INFRA] Exposing JIRA issue component types at GitHub PRs

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
## What changes were proposed in this pull request?

This PR aims to expose JIRA issue component types at GitHub PRs.

## How was this patch tested?

Manual.
```
$ export GITHUB_OAUTH_KEY=...
$ export JIRA_PASSWORD=...
$ export GITHUB_API_BASE='https://api.github.com/repos/your-id/spark'
$ dev/github_jira_sync.py
```

Please note that the existing script will raise the following exceptions if your repo has less than 100 PRs. This will be handled at #24874 .
```
Traceback (most recent call last):
  File "dev/github_jira_sync.py", line 139, in <module>
    jira_prs = get_jira_prs()
  File "dev/github_jira_sync.py", line 83, in get_jira_prs
    link_header = filter(lambda k: k.startswith("Link"), page.info().headers)[0]
IndexError: list index out of range
```
That is beyond the scope of this PR.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
I merged #24874 to unblock this one.

@JoshRosen, I believe this script should be re-ran after this PR merged.

**Comment by dongjoon-hyun** [comment]
Thank you, @HyukjinKwon ! I rebased this PR to the master for reviewers to get a non-error environment.

**Review by felixcheung — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Thank you, @felixcheung and @HyukjinKwon .
Merged to master.

**Pull request closed without merging**
